stdlib/public/ExternalGenericMetadataBuilder/ExternalGenericMetadataBuilder.cpp

```diff
@@ -0,0 +1,2444 @@
+void ExternalGenericMetadataBuilderContext<Runtime>::cacheDescriptor(
+    Buffer<const ContextDescriptor> descriptorBuffer) {
+  auto desc = descriptorBuffer.ptr;
+  auto typeDesc = dyn_cast<const TypeContextDescriptor>(desc);
+  if (!typeDesc) {
+    return;
+  }
+
+  // This produces a Type -> {Structure|Class|Enum}
+  Demangle::Demangler Dem;
+  auto demangleNode = buildDemanglingForContext(descriptorBuffer, Dem);
+  if (auto *error = demangleNode.getError()) {
+    auto symbolInfo = readerWriter->getSymbolInfo(descriptorBuffer);
+    LOG(LogLevel::Warning,
+        "Could not build demangling for context %s (%s + %" PRIu64 "): %s",
+        symbolInfo.symbolName.c_str(), symbolInfo.libraryName.c_str(),
+        symbolInfo.pointerOffset, error->cStr());
+    return;
+  }
+
+  // ... which needs to be wrapped in Global -> NominalTypeDescriptor -> * to
+  // match what we cache during dylib reading.
+  auto global = wrapNode(Dem, *demangleNode.getValue(), Node::Kind::Global,
+                         Node::Kind::NominalTypeDescriptor);
+
+  auto mangling = Demangle::mangleNode(global);
+  if (!mangling.isSuccess()) {
+    auto symbolInfo = readerWriter->getSymbolInfo(descriptorBuffer);
+    LOG(LogLevel::Warning,
+        "Could not build demangling for context %s (%s + %" PRIu64 "): %s",
+        symbolInfo.symbolName.c_str(), symbolInfo.libraryName.c_str(),
+        symbolInfo.pointerOffset, getNodeTreeAsString(global).c_str());
+    return;
+  }
+
+  LOG(LogLevel::Info, "Found descriptor %s", mangling.result().c_str());
+
+  auto [iterator, didInsert] = mangledNominalTypeDescriptorMap.try_emplace(
+      mangling.result(),
+      descriptorBuffer.template cast<const TypeContextDescriptor>());
+
+  if (!didInsert) {
+    auto [name, oldTypeDesc] = *iterator;
+    LOG(LogLevel::Warning,
+        "Descriptor for %s already exists - old: %p - new: %p", name.c_str(),
+        oldTypeDesc.ptr, typeDesc);
+  }
+}
+
+// Perform prep tasks on file, add it to machOFiles, and take ownership of the
+// object.
+template <typename Runtime>
+void ExternalGenericMetadataBuilderContext<Runtime>::addImage(MachOFile *file) {
+  // Replace the path with the library's install name if possible.
+  auto installName = getInstallName(file->objectFile.get());
+  if (installName.length() > 0)
+    file->path = installName;
+
+  populateMachOSymbols(file);
+  populateMachOFixups(file);
+  readMachOSections(file);
+  machOFiles.emplace_back(file);
+  machOFilesByPath.insert({file->path, file});
+}
+
+template <typename Runtime>
+void ExternalGenericMetadataBuilderContext<Runtime>::populateMachOSymbols(
+    MachOFile *file) {
+  // Fill out library names.
+  for (auto &command : file->objectFile->load_commands()) {
+    if (command.C.cmd == llvm::MachO::LC_LOAD_DYLIB ||
+        command.C.cmd == llvm::MachO::LC_LOAD_WEAK_DYLIB ||
+        command.C.cmd == llvm::MachO::LC_REEXPORT_DYLIB ||
+        command.C.cmd == llvm::MachO::LC_LOAD_UPWARD_DYLIB) {
```

Comment:
Do we need `LC_LAZY_LOAD_DYLIB`?

Reply:
It seems not. `LC_LAZY_LOAD_DYLIB` seems to be obsolete, and this matches the behavior of dyld.